packages/pigeon/lib/cpp_generator.dart

```diff
@@ -64,6 +70,24 @@ class CppOptions {
   }
 }
 
+/// Class that manages all Cpp code generation.
+class CppGenerator extends Generator<CppOptions> {
+  /// Instantiates a Cpp Generator.
+  CppGenerator();
+
+  /// Generates Cpp files with specified [CppOptions]
+  @override
+  void generate(CppOptions languageOptions, Root root, StringSink sink,
+      {required FileType fileType}) {
```

Comment:
Yea, this looks good to me.  I'd just pull `fileType` into the CppOptions since it isn't applicable to all Generators.

Reply:
It applies to two already, and will apply to three when we add Linux, so whatever solution we have shouldn't be specific to C++.

If we put it in options, we won't be able to pass the same immutable options to reach generate call, which undermines the idea that all calls to generate need to happen with identical options (so that the generated code matches).